packages/headless/src/api/analytics/analytics.ts

```diff
@@ -56,6 +58,7 @@ export class AnalyticsProvider implements SearchPageClientProvider {
   public getSearchUID() {
     return (
       this.state.search?.response.searchUid ||
+      this.state.recommendation?.searchUid ||
```

Comment:
I would need to read and understand the whole class, but this line sandwiched between `search.reponse` feels out-of-place.

I see that the `AnalyticsProvider` class implements `SearchPageClientProvider` (SearchPage).
Since Recommendation is powered by an independent engine (not a SearchApp engine), should it have it's own analytics provider?

Reply:
Yeahhhhh we could.... 🤔 

That would be the only difference though as far as I can tell. The rest would be an exact carbon copy.

Another approach that I think would be better/less rework would be to inverse the dependency a bit, and have the analytics actions "push" the ID to use, instead of having the SearchPageClientProvider pull from the state, and have it decide between "recommendation or search section".

Reply:
Opened a spike here : https://coveord.atlassian.net/browse/KIT-367